src/deploy/functions/prepare.ts

```diff
@@ -180,6 +182,126 @@ export async function prepare(
   await promptForFailurePolicies(options, matchingBackend, haveBackend);
   await promptForMinInstances(options, matchingBackend, haveBackend);
   await backend.checkAvailability(context, wantBackend);
+  await validateSecrets(matchingBackend);
+  await ensureSecretAccess(matchingBackend);
+}
+
+/**
+ * Validate secret environment variables setting, if any.
+ * A bad secret configuration can lead to a significant delay in function deploys.
+ *
+ * If validation fails for any secret config, throws a FirebaseError.
+ */
+export async function validateSecrets(b: backend.Backend) {
```

Comment:
nit: return type

Reply:
After reading few more TS tutorirals, I don't think typing `Promise<void>` adds to much value here, especially since TS can easily figure it out implicitly. WDYT?

Reply:
Sure that's fine with me, but the GitHub actions linter thing still throws a warning